qemu/tests/libvirt_host_model_test.py

```diff
@@ -0,0 +1,175 @@
+    feature_list = cpu_xml.get_feature_list()
+    cpu_model = cpu_xml.model
+    cpu_features = {}
+    for i in range(0, len(feature_list)):
+        feature_name = cpu_xml.get_feature(i).get('name')
+        feature_policy = cpu_xml.get_feature(i).get('policy')
+        if feature_policy == "require":
+            feature_policy = "on"
+        elif feature_policy == "disable":
+            feature_policy = "off"
+        cpu_features[feature_name] = feature_policy
+    cpu_info = {}
+    cpu_info["model"] = cpu_model
+    cpu_info["features"] = cpu_features
+    return cpu_info
+
+
+def compare_cpu_info(test, params):
+    """
+    Compare flags between qemu cli and libvirt dumpxml with 'host-model'
+    :param test: QEMU test object
+    :param params: Dictionary with the test parameters
+    :return: True or False
+    """
+    vm_arch = params["vm_arch_name"]
+    machine = params.get('machine_type')
+    name = uuid.uuid4().hex
+    try:
+        path.find_command("virsh")
+    except path.CmdNotFoundError:
+        test.cancel("Virsh executable not set or found on path")
+
+    xml = """
+            <domain type='kvm'>
+                <name>%s</name>
+                <memory>1</memory>
+                <os>
+                    <type arch='%s' machine='%s'>hvm</type>
+                </os>
+                <cpu mode='host-model'/>
+            </domain>
+          """ % (name, vm_arch, machine)
+    xml_file = os.path.join(data_dir.get_tmp_dir(), "temp_xml_for_cpu")
+    with open(xml_file, "w") as f:
+        f.write(xml)
+    try:
+        logging.info("Get cpu model and features from virsh")
+        virsh.define(xml_file)
+        virsh.start(name)
+    except Exception as err:
+        test.cancel(err)
+    else:
+        qemu_cpu_info = cpu.get_cpu_info_from_virsh_qemu_cli(name)
+        libvirt_cpu_info = get_cpu_info_from_dumpxml(name)
+        params["qemu_cpu_info"] = qemu_cpu_info
+        params["libvirt_cpu_info"] = libvirt_cpu_info
+
+        cpu_model_qemu = qemu_cpu_info["model"]
+        cpu_model_libvirt = libvirt_cpu_info["model"]
+        params["qemu_proc_cpu_flags"] = qemu_cpu_info["flags"]
+        if cpu_model_qemu != cpu_model_libvirt:
+            logging.error("mismatch cpu model bwteen qemu %s and libvirt %s",
+                          cpu_model_qemu, cpu_model_libvirt)
+            return False
+        params["cpu_model"] = cpu_model_qemu
+        qemu_cpu_flags = cpu.parse_qemu_cpu_flags(qemu_cpu_info['flags'])
+        libvirt_cpu_flags = libvirt_cpu_info['features']
+        qemu_cpu_flags = extend_flags_patterns(qemu_cpu_flags)
+        exclude_map = eval(params.get('exclude_map', '{}'))
+        check_exclude = False
+        if cpu_model_qemu in exclude_map.keys():
+            exclude_map_flags = exclude_map[cpu_model_qemu]
+            check_exclude = True
+        miss_flags = []
+        mismatch_flags = []
+        result_bool = True
+        for flag in libvirt_cpu_flags.keys():
```

Comment:
Any chance a flag in qemu_cpu_flags but not in libvirt_cpu_flags?

Reply:
> Any chance a flag in qemu_cpu_flags but not in libvirt_cpu_flags?

Seems no since we get all information by booting a boot via libvirt, so all flags are supported to be configured in xml.

Reply:
@huangyum  Any more comments ?

Reply:
> > Any chance a flag in qemu_cpu_flags but not in libvirt_cpu_flags?
> 
> Seems no since we get all information by booting a boot via libvirt, so all flags are supported to be configured in xml.

Not sure about it. Would it be a libvirt or qemu bug that cause qemu flags more than libvirt flags?

Reply:
And once that happens, we will be not aware because we don't check that.

Reply:
> And once that happens, we will be not aware because we don't check that.

This situation can't happen since once there is an unexpected flag that libvirt can't configure, the libvirt won't start with an error. 
And also confirm this with libvirt team, so I think we can ignore this situation, thanks.

Reply:
Sure, your call.